### DISCUSSION TITLE
queue.add() hanging

### DISCUSSION THREAD

**Discussion by aryeharmon** [OptimalBits/bull#2559] [Q&A]
i have a few hundred bull queues running in a single app. i find that often when adding jobs: 
`await queue.add() `
hangs and never completes. i am trying to figure out where this is coming from. i do not see the redis server cpu or memorry full at all so i am looking for ideas about where this might be coming from and what can cause this sort of problem. any ideas?

**Comment by manast** [comment]
It could be simply that your connection to Redis is down, configure your Queue instance so that it fails fast as explained here to avoid hanging: https://docs.bullmq.io/patterns/failing-fast-when-redis-is-down

**Reply by aryeharmon** [reply]
does not look like this exists in the normal bull npm package, is this correct?

**Reply by maxxer** [reply]
Same here, did you solve it? Thanks
Connection to Redis seems fine on my side

**Reply by maxxer** [reply]
Apparently still a Redis connection issue, as I'm using Vercel KV and it's TLS enabled

**Reply by manast** [reply]
Yes, but if you follow the instructions [here](https://docs.bullmq.io/patterns/failing-fast-when-redis-is-down) it should not hang forever, just throw an exception.

**Reply by maxxer** [reply]
thanks. it's unclear what docs are related to bullmq only
